lib/bundler/definition.rb

```diff
@@ -218,15 +218,42 @@ def index
           dependency_names -= pinned_spec_names(source.specs)
           dependency_names.push(*source.unmet_deps).uniq!
         end
+
+        memoize = proc do |&work|
+          ret = memoize
+          proc do
+            if memoize.equal?(ret)
+              ret = work.call
+            else
+              ret
+            end
+          end
+        end
```

Comment:
Is there an extremely good reason to make this a prod instead of a method? Procs are sooo much slower than method calls. Also, procs are much harder to read and understand than methods... like how i can't even tell if this has to be a prof instead of a method. 😝